### PR TITLE
Storage Documentation

### PR DESCRIPTION
**Pull request by beaushinkle**
This PR adds a section to the README detailing our storage architecture. It should help give users an idea of what files we're generating on their computers and how to understand them.

tagging @nkuba for review

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -197,6 +197,57 @@ go test ./...
...
+│   │   ├── ypub_6Z2Yy4ZH
+│   │   │   ├── 1692
+│   │   │   ├── 223
+│   │   │   ├── 48
+│   │   │   └── 8982
+│   │   └── zpub_VDaZuQT9
+│   │       ├── 112
+│   │       ├── 6311
+│   │       ├── 6999
+│   │       └── 8559
```
Comment (nkuba) [inline README.adoc:227]: Isn't it just one file per address stored with the latest used index?

image

File: README.adoc [hunk 2]
```diff
@@ -197,6 +197,57 @@ go test ./...
...
+
+The `archive`, `current`, and `snapshot` directories are for storing key
+material. The `current` directory represents currently active keeps, and the
+archive directory contains closed and terminated keeps. The snapshot directory
+exists to guarentee persistence of the key material before it is published on
+chain.
+
+The `bitcoin/derivation_indexes` directory is used to store used derivation
+indexes for extended public keys. Each folder represents a truncated public key
+and each 0-byte file within those folders represents a used index.
```
Comment (nkuba) [inline README.adoc:248]: We should mention in liquidation recovery instruction how an operator can configure desired starting index.

Comment (nkuba) [reply README.adoc:248]: But let's do it in a follow-up PR.

File: README.adoc (hunk 1 above)
Comment (beaushinkle) [reply README.adoc:227]: Oh, yup! I forgot that we changed it to that and grabbed the folder structure from the original PR

**Review by nkuba — APPROVED** [review]

**Pull request merged**